tools/validate_spec.py

```diff
@@ -3,261 +3,286 @@
+from types import TracebackType
+from typing import IO, Dict, Iterator, List, Optional, Sequence, Type, Union
 
-from rflx.error import RecordFluxError
 from rflx.identifier import ID
 from rflx.pyrflx import PyRFLX, TypeValue
 from rflx.pyrflx.typevalue import MessageValue
 
 
 def cli(argv: List[str]) -> Union[int, str]:
-    parser = argparse.ArgumentParser()
-    subparsers = parser.add_subparsers(dest="subcommand")
-    subparsers.required = True
+    parser = argparse.ArgumentParser(
+        description="validate specification against a set of known valid or invalid messages"
+    )
 
-    parser_verify = subparsers.add_parser(
-        "verify", help="check specification against a set of known valid or invalid examples"
+    parser.add_argument(
+        "-s", "--specification", type=Path, help="specification file", required=True
     )
 
-    parser_verify.add_argument(
-        "-d-valid",
-        "--valid-directory",
-        type=Path,
-        help=("absolute path to directory with known valid examples"),
-        default=None,
+    parser.add_argument(
+        "-m",
+        "--message-identifier",
+        type=str,
+        help="identifier of the top-level message: <PackageName>::<MessageName>",
+        required=True,
     )
 
-    parser_verify.add_argument(
-        "-d-invalid",
-        "--invalid-directory",
+    parser.add_argument(
+        "-v",
+        "--directory-valid",
         type=Path,
-        help=("absolute path to directory with known valid examples"),
+        help="path to directory with known valid examples",
         default=None,
     )
 
-    parser_verify.add_argument(
-        "-m",
-        "--message-identifier",
-        type=str,
-        help=("identifier of the top-level message: <PackageName>.<MessageName>"),
-        required=True,
+    parser.add_argument(
+        "-i",
+        "--directory-invalid",
+        type=Path,
+        help="path to directory with known valid examples",
+        default=None,
     )
 
-    parser_verify.add_argument(
+    parser.add_argument(
         "-o",
         "--json-output",
         type=Path,
-        help=("identifier of the top-level message: <PackageName>.<MessageName>"),
+        help="path to output file - file must not exist",
         default=None,
     )
 
-    parser_verify.add_argument(
+    parser.add_argument(
         "--abort-on-error",
         action="store_true",
-        help=("abort with exitcode 1 if a message cannot be verified correctly"),
+        help=(
+            "abort with exitcode 1 if a message is classified "
+            "as a false positive or false negative"
+        ),
     )
 
-    parser_verify.add_argument("-s", "--specification", type=Path, help=("specification file"))
+    args = parser.parse_args(argv[1:])
+    if args.directory_invalid is None and args.directory_invalid is None:
+        parser.error("must provide directory with valid and/or invalid messages")
```

Comment:
`args.directory_invalid` is checked two times.

Reply:
Fixed in a4e2b48